File: Reclamation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups ;

/**
 * reclamation
 *
 * @ORM\Table(name="reclamation")
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\ReclamationRepository")
 */
class Reclamation
{
    /**
     * @var int
     *
     * @ORM\Column(name="idRec", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("posts:read")
     */
    private $idrec;

    /**
     * @var int
     *
     * @ORM\Column(name="idJoueur", type="integer", nullable=false)
     *  @Groups("posts:read")
     */
    private $idjoueur;

    /**
     * @var string
     *
     * @ORM\Column(name="SujetRec", type="string", length=75, nullable=false)
     *  @Groups("posts:read")
     */
    private $sujetrec;

    /**
     * @var string
     *
     * @ORM\Column(name="DescriptionRec", type="string", length=250, nullable=false)
     *  @Groups("posts:read")
     */
    private $descriptionrec;

    /**
     * @var string
     *
     * @ORM\Column(name="StatusRec", type="string", length=55, nullable=false)
     *  @Groups("posts:read")
     */
    private $statusrec;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="DateRec", type="date", nullable=false)
     *  @Groups("posts:read")
     */
    private $daterec;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="DateTraitement", type="date", nullable=true, options={"default"="NULL"})
     *  @Groups("posts:read")
     */
    private $datetraitement = 'NULL';

    /**
     * @var string|null
     *
     * @ORM\Column(name="NomPrenomCoach", type="string", length=50, nullable=true, options={"default"="NULL"})
     *  @Groups("posts:read")
     */
    private $nomprenomcoach = 'NULL';

    /**
     * @var string
     *
     * @ORM\Column(name="NomJoueur", type="string", length=255, nullable=false)
     *  @Groups("posts:read")
     */
    private $nomjoueur;

    /**
     * @var string
     *
     * @ORM\Column(name="PrenomJoueur", type="string", length=255, nullable=false)
     *  @Groups("posts:read")
     */
    private $prenomjoueur;

    /**
     * @var string
     *
     * @ORM\Column(name="EmailJoueur", type="string", length=255, nullable=false)
     *  @Groups("posts:read")
     */
    private $emailjoueur;

    /**
     * @var string
     *
     * @ORM\Column(name="TlJoueur", type="string", length=12, nullable=false)
     *  @Groups("posts:read")
     */
    private $tljoueur;

    /**
     * @var string|null
     *
     * @ORM\Column(name="imgRec", type="string", length=255, nullable=true, options={"default"="NULL"})
     *  @Groups("posts:read")
     */
    private $imgrec = 'NULL';

    /**
     * @var int
     *
     * @ORM\Column(name="Opened", type="integer", nullable=false)
     *  @Groups("posts:read")
     */
    private $opened = '0';

    public function getIdrec(): ?int
    {
        return $this->idrec;
    }

    public function getIdjoueur(): ?int
    {
        return $this->idjoueur;
    }

    public function setIdjoueur(int $idjoueur): self
    {
        $this->idjoueur = $idjoueur;

        return $this;
    }

    public function getSujetrec(): ?string
    {
        return $this->sujetrec;
    }

    public function setSujetrec(string $sujetrec): self
    {
        $this->sujetrec = $sujetrec;

        return $this;
    }

    public function getDescriptionrec(): ?string
    {
        return $this->descriptionrec;
    }

    public function setDescriptionrec(string $descriptionrec): self
    {
        $this->descriptionrec = $descriptionrec;

        return $this;
    }

    public function getStatusrec(): ?string
    {
        return $this->statusrec;
    }

    public function setStatusrec(string $statusrec): self
    {
        $this->statusrec = $statusrec;

        return $this;
    }

    public function getDaterec(): ?\DateTimeInterface
    {
        return $this->daterec;
    }

    public function setDaterec(\DateTimeInterface $daterec): self
    {
        $this->daterec = $daterec;

        return $this;
    }

    public function getDatetraitement(): ?\DateTimeInterface
    {
            return null;
        if ($this->Datetraitement !== null) {
            return $this->Datetraitement;
        }

    }

    public function setDatetraitement(?\DateTimeInterface $datetraitement): self
    {
        $this->datetraitement = $datetraitement;

        return $this;
    }

    public function getNomprenomcoach(): ?string
    {
        return $this->nomprenomcoach;
    }

    public function setNomprenomcoach(?string $nomprenomcoach): self
    {
        $this->nomprenomcoach = $nomprenomcoach;

        return $this;
    }

    public function getNomjoueur(): ?string
    {
        return $this->nomjoueur;
    }

    public function setNomjoueur(string $nomjoueur): self
    {
        $this->nomjoueur = $nomjoueur;

        return $this;
    }

    public function getPrenomjoueur(): ?string
    {
        return $this->prenomjoueur;
    }

    public function setPrenomjoueur(string $prenomjoueur): self
    {
        $this->prenomjoueur = $prenomjoueur;

        return $this;
    }

    public function getEmailjoueur(): ?string
    {
        return $this->emailjoueur;
    }

    public function setEmailjoueur(string $emailjoueur): self
    {
        $this->emailjoueur = $emailjoueur;

        return $this;
    }

    public function getTljoueur(): ?string
    {
        return $this->tljoueur;
    }

    public function setTljoueur(string $tljoueur): self
    {
        $this->tljoueur = $tljoueur;

        return $this;
    }

    public function getImgrec()
    {
        return $this->imgrec;
    }

    public function setImgrec( $imgrec)
    {
        $this->imgrec = $imgrec;

        return $this;
    }

    public function getOpened(): ?int
    {
        return $this->opened;
    }

    public function setOpened(int $opened): self
    {
        $this->opened = $opened;

        return $this;
    }


}

File: src/com/menu/coheal/Models/Reclamation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entites;



/**
 *
 * @author yassin
 */
public class Reclamation {
    
    private int idRec, idJoueur;
    private String SujetRec, DescriptionRec, StatusRec, PrioriteRec;
    private String NomJoueur, PrenomJoueur, EmailJoueur, TlJoueur, NomPrenomCoach, imgRec;
    private String DateRec, DateTraitement;

    public Reclamation() {
    }

    
    public Reclamation(int idRec, int idJoueur, String SujetRec, String DescriptionRec, String StatusRec, String PrioriteRec, String NomJoueur, String PrenomJoueur, String EmailJoueur, String TlJoueur, String NomPrenomCoach, String imgRec, String DateRec, String DateTraitement) {
        this.idRec = idRec;
        this.idJoueur = idJoueur;
        this.SujetRec = SujetRec;
        this.DescriptionRec = DescriptionRec;
        this.StatusRec = StatusRec;
        this.PrioriteRec = PrioriteRec;
        this.NomJoueur = NomJoueur;
        this.PrenomJoueur = PrenomJoueur;
        this.EmailJoueur = EmailJoueur;
        this.TlJoueur = TlJoueur;
        this.NomPrenomCoach = NomPrenomCoach;
        this.imgRec = imgRec;
        this.DateRec = DateRec;
        this.DateTraitement = DateTraitement;
    }

    public Reclamation(int idJoueur, String SujetRec, String DescriptionRec, String StatusRec, String PrioriteRec, String NomJoueur, String PrenomJoueur, String EmailJoueur, String TlJoueur, String NomPrenomCoach, String imgRec, String DateRec, String DateTraitement) {
        this.idJoueur = idJoueur;
        this.SujetRec = SujetRec;
        this.DescriptionRec = DescriptionRec;
        this.StatusRec = StatusRec;
        this.PrioriteRec = PrioriteRec;
        this.NomJoueur = NomJoueur;
        this.PrenomJoueur = PrenomJoueur;
        this.EmailJoueur = EmailJoueur;
        this.TlJoueur = TlJoueur;
        this.NomPrenomCoach = NomPrenomCoach;
        this.imgRec = imgRec;
        this.DateRec = DateRec;
        this.DateTraitement = DateTraitement;
    }
    
        public Reclamation(String SujetRec, String DescriptionRec, String DateRec) {
        this.SujetRec = SujetRec;
        this.DescriptionRec = DescriptionRec;
        this.DateRec = DateRec;
    }

    public int getIdRec() {
        return idRec;
    }

    public void setIdRec(int idRec) {
        this.idRec = idRec;
    }

    public int getIdJoueur() {
        return idJoueur;
    }

    public void setIdJoueur(int idJoueur) {
        this.idJoueur = idJoueur;
    }

    public String getSujetRec() {
        return SujetRec;
    }

    public void setSujetRec(String SujetRec) {
        this.SujetRec = SujetRec;
    }

    public String getDescriptionRec() {
        return DescriptionRec;
    }

    public void setDescriptionRec(String DescriptionRec) {
        this.DescriptionRec = DescriptionRec;
    }

    public String getStatusRec() {
        return StatusRec;
    }

    public void setStatusRec(String StatusRec) {
        this.StatusRec = StatusRec;
    }

    public String getPrioriteRec() {
        return PrioriteRec;
    }

    public void setPrioriteRec(String PrioriteRec) {
        this.PrioriteRec = PrioriteRec;
    }

    public String getNomJoueur() {
        return NomJoueur;
    }

    public void setNomJoueur(String NomJoueur) {
        this.NomJoueur = NomJoueur;
    }

    public String getPrenomJoueur() {
        return PrenomJoueur;
    }

    public void setPrenomJoueur(String PrenomJoueur) {
        this.PrenomJoueur = PrenomJoueur;
    }

    public String getEmailJoueur() {
        return EmailJoueur;
    }

    public void setEmailJoueur(String EmailJoueur) {
        this.EmailJoueur = EmailJoueur;
    }

    public String getTlJoueur() {
        return TlJoueur;
    }

    public void setTlJoueur(String TlJoueur) {
        this.TlJoueur = TlJoueur;
    }

    public String getNomPrenomCoach() {
        return NomPrenomCoach;
    }

    public void setNomPrenomCoach(String NomPrenomCoach) {
        this.NomPrenomCoach = NomPrenomCoach;
    }

    public String getImgRec() {
        return imgRec;
    }

    public void setImgRec(String imgRec) {
        this.imgRec = imgRec;
    }

    public String getDateRec() {
        return DateRec;
    }

    public void setDateRec(String DateRec) {
        this.DateRec = DateRec;
    }

    public String getDateTraitement() {
        return DateTraitement;
    }

    public void setDateTraitement(String DateTraitement) {
        this.DateTraitement = DateTraitement;
    }
    
    
    
    
    
}
